core/tauri/src/updater/core.rs

```diff
@@ -826,173 +832,69 @@ fn copy_files_and_run<R: Read + Seek>(
+        .nsis_args()
+        .iter()
+        .map(OsStr::new)
+      );
     } else if found_path.extension() == Some(OsStr::new("msi")) {
-      if with_elevated_task {
-        if let Some(bin_name) = current_exe()
-          .ok()
-          .and_then(|pb| pb.file_name().map(|s| s.to_os_string()))
-          .and_then(|s| s.into_string().ok())
-        {
-          let product_name = bin_name.replace(".exe", "");
-
-          // Check if there is a task that enables the updater to skip the UAC prompt
-          let update_task_name = format!("Update {product_name} - Skip UAC");
-          if let Ok(output) = Command::new("schtasks")
-            .arg("/QUERY")
-            .arg("/TN")
-            .arg(update_task_name.clone())
-            .output()
-          {
-            if output.status.success() {
-              // Rename the MSI to the match file name the Skip UAC task is expecting it to be
-              let temp_msi = tmp_dir.with_file_name(bin_name).with_extension("msi");
-              Move::from_source(&found_path)
-                .to_dest(&temp_msi)
-                .expect("Unable to move update MSI");
-              let exit_status = Command::new("schtasks")
-                .arg("/RUN")
-                .arg("/TN")
-                .arg(update_task_name)
-                .status()
-                .expect("failed to start updater task");
-
-              if exit_status.success() {
-                // Successfully launched task that skips the UAC prompt
-                exit(0);
-              }
-            }
-            // Failed to run update task. Following UAC Path
-          }
-        }
-      }
-
-      // we need to wrap the current exe path in quotes for Start-Process
-      let mut current_executable = std::ffi::OsString::new();
-      current_executable.push("\"");
-      current_executable.push(dunce::simplified(&current_exe()?));
-      current_executable.push("\"");
-
-      let mut msi_path = std::ffi::OsString::new();
-      msi_path.push("\"\"\"");
-      msi_path.push(&found_path);
-      msi_path.push("\"\"\"");
-
-      let installer_args = [
-        config.tauri.updater.windows.install_mode.msiexec_args(),
-        config
-          .tauri
-          .updater
-          .windows
-          .installer_args
-          .iter()
-          .map(AsRef::as_ref)
-          .collect::<Vec<_>>()
-          .as_slice(),
-      ]
-      .concat();
-
-      // run the installer and relaunch the application
-      let mut powershell_cmd = Command::new(powershell_path);
-
-      powershell_cmd
-        .args(["-NoProfile", "-WindowStyle", "Hidden"])
-        .args([
-          "Start-Process",
-          "-Wait",
-          "-FilePath",
-          "$Env:SYSTEMROOT\\System32\\msiexec.exe",
-          "-ArgumentList",
-        ])
-        .arg("/i,")
-        .arg(&msi_path)
-        .arg(format!(", {}, /promptrestart;", installer_args.join(", ")))
-        .arg("Start-Process")
-        .arg(current_executable);
```

Comment:
this logic will try to re-run the application after MSI update is done, we can't remove this logic either, or it will be considered breaking change.